cloud/vsphere/clusteractuator.go

```
@@ -61,6 +70,91 @@ func (vc *ClusterActuator) Reconcile(cluster *clusterv1.Cluster) error {
+		return vsphereconfig.NotReady, nil
+	}
+	// Currently we support only a single master thus the below assumption
+	// Once we start supporting multiple masters, the kubeconfig needs to
+	// be generated differently, with the URL from the LB endpoint
+	master := masters[0]
+	kubeconfig, err := vsphereutils.GetKubeConfig(cluster, master)
+	if err != nil || kubeconfig == "" {
+		glog.Infof("[cluster-actuator] error retrieving kubeconfig for target cluster, will requeue")
+		return vsphereconfig.NotReady, &clustererror.RequeueAfterError{RequeueAfter: constants.RequeueAfterSeconds}
+	}
+	kconfigFile, err := vsphereutils.CreateTempFile(kubeconfig)
+	if err != nil {
+		return vsphereconfig.NotReady, err
+	}
+	clientConfig, err := clientcmd.BuildConfigFromFlags("", kconfigFile)
+	if err != nil {
+		glog.Infof("[cluster-actuator] error creating client config for target cluster [%s], will requeue", err.Error())
+		return vsphereconfig.NotReady, &clustererror.RequeueAfterError{RequeueAfter: constants.RequeueAfterSeconds}
+	}
+	clientSet, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		glog.Infof("[cluster-actuator] error creating clientset for target cluster [%s], will requeue", err.Error())
+		return vsphereconfig.NotReady, &clustererror.RequeueAfterError{RequeueAfter: constants.RequeueAfterSeconds}
+	}
+	_, err = clientSet.Core().Nodes().List(metav1.ListOptions{})
+	if err != nil {
+		glog.Infof("[cluster-actuator] target cluster API not yet ready [%s], will requeue", err.Error())
+		return vsphereconfig.NotReady, &clustererror.RequeueAfterError{RequeueAfter: constants.RequeueAfterSeconds}
+	}
+	return vsphereconfig.Ready, nil
+}
+
+func (vc *ClusterActuator) updateClusterAPIStatus(cluster *clusterv1.Cluster, newStatus vsphereconfig.APIStatus) error {
+	oldProviderStatus, err := vsphereutils.GetClusterProviderStatus(cluster)
+	if err != nil {
+		return err
+	}
+	currentAPIStatus, err := vc.getClusterAPIStatus(cluster)
+	if err != nil {
+		return err
```

Do you really want to return the error here? If the status is `NotReady`, wouldn't it be better to update the status with this condition?